src/osd/PG.cc

```diff
@@ -4140,6 +4140,8 @@ int PG::build_scrub_map_chunk(
 
   map.valid_through = info.last_update;
 
+  osr->flush();
```

Comment:
Hmm, this is another potentially-blocking flush we’re adding in a worker thread. Dislike. :/

More to the point, I thought rollback objects were in a special namespace or region. I don’t even remember if they get scrubbed, but if they do we can probably conditionalize this on it being in their space. Though that is pretty gross. Hrmm.

Reply:
AFAICS the flush is needed.. pretty sure it is a bug that is hasn't been there.  My guess is that because the scrub range has blocked io it is very difficult to hit an unapplied update, but it is still possible.

The real solution is to make the ObjectStore(s) not require a flush at all, but that requires a bit of work in bluestore and a lot more work in filestore.  I think we should kick that one down teh road a bit more until the weight of deployments is on bluestore so that we can suffer a filestore regression in order to streamline/simplify the interface.

Reply:
Well, yeah, that does make sense. Did we previously use a sequencer for listing operations, perhaps?

Or actually, we probably wait on the primary until nothing in the scrub range has any operations in progress, thanks to the ObjectContext locks? So I guess there's still a race but it's pretty unlikely since it involves extra network round-trips anyway?

The osr is at least per-PG so it doesn't force extra waits, right?